src/plugins/layout/stateprovider.ts

```diff
@@ -151,25 +222,26 @@ export function createIgnoreLayoutState(node: Node): IIgnoreLayoutState {
 }
 
 export function createGenericLayoutState(node: Node, tag: string): ILayoutState {
-  let layoutState: ILayoutState = {
-    index: getNodeIndex(node),
+  let layoutIndex = getNodeIndex(node);
+  layoutStates[layoutIndex] = {
+    index: layoutIndex,
     parent: getNodeIndex(node.parentNode),
     previous: getNodeIndex(node.previousSibling),
     next: getNodeIndex(node.nextSibling),
     source: null,
     action: null,
     tag
   };
-  return layoutState;
+
+  return layoutStates[layoutIndex];
```

Comment:
Since you made a function for it, it would make sense to use getLayoutState(layoutIndex) here

Reply:
I'm not sure if I understand what you are referring to here. This is a set, not a get here.